conf/spark-defaults.conf.template

```diff
@@ -19,9 +19,16 @@
 # This is useful for setting default environmental settings.
 
 # Example:
-# spark.master                     spark://master:7077
-# spark.eventLog.enabled           true
-# spark.eventLog.dir               hdfs://namenode:8021/directory
-# spark.serializer                 org.apache.spark.serializer.KryoSerializer
-# spark.driver.memory              5g
-# spark.executor.extraJavaOptions  -XX:+PrintGCDetails -Dkey=value -Dnumbers="one two three"
+# spark.master                          spark://master:7077
+# spark.eventLog.enabled                true
+# spark.eventLog.dir                    hdfs://namenode:8021/directory
+# spark.serializer                      org.apache.spark.serializer.KryoSerializer
+# spark.driver.memory                   5g
+# spark.executor.extraJavaOptions       -XX:+PrintGCDetails -Dkey=value -Dnumbers="one two three"
+
+#spark.ui.allowFramingFrom              https://example.com/
+#spark.ui.xXssProtection.enabled        1; mode=block
+#spark.ui.xContentType.options          nosniff
+
+#Enable below only when Spark is running on HTTPS
+#spark.ui.strictTransportSecurity.age   max-age=31536000
```

Comment:
On a similar note does max-age need to be part of the user-supplied string or just the age? in what units?

Reply:
Yes, "max-age" need to be part of user-supplied string. Possible values can be one from the below set, default one mentioned in template is for optimum secure value commonly used.

Strict-Transport-Security: {max-age=\<expire-time\> | max-age=\<expire-time\>; includeSubDomains | max-age=\<expire-time\>; preload}


Value is in delta-seconds. More here https://tools.ietf.org/html/rfc6797#section-6.1.1

Reply:
I'd just remove '.age' then, likewise. It's the very value of the STS header

Reply:
I understand that the "max-age" part is common across all values but don't want to tamper the value part for ease and any future compatibility. I will rather remove ".age" to avoid confusion arising out of this.